### ISSUE TITLE
Need Help Getting Cutout to work Properly

### ISSUE THREAD

**Issue by pGit1** [uoguelph-mlrg/Cutout#6]
Hi @TDeVries and @gwtaylor ! This paper is amazing(!!!!) and I've been using it's insights ever since it came out. Although I feel I am not getting the same benefits everyone else is getting when applying this regularization technique to my training process. I have a few questions to clear some cobwebs in my head regarding the technique:

1. How important is per channel normalization to getting cutout to work properly? Is dividing by 255 a bad way to normalize images? What is a go-to image normalization approach?

2. When should cutout be applied? Should it be applied **before OR after** image normalization?

3. If I have other augmentations happening to my images, should cutout be applied **before OR after** my other augmentations?

Thank you!!

**Comment by TDeVries** [comment]
Hello, glad you liked the paper!

For a lot of these things I didn't actually run any experiments to see if specific details or order of operations was important since everything seemed to work on the first try, so we just went with that. I can comment a bit though: 

1. In the paper we subtract the per-channel mean and divide by the per-channel standard deviation, and then apply a 0 mask in an attempt to somewhat preserve batch statistics (since the mean of the data is now 0).  If you divide by 255 and then apply a 0 mask for cutout it might slightly shift the data distribution from training and testing, so it may not work as well (this is speculation, I haven't tested it). An alternative approach to applying a 0 mask is to use random noise from a uniform distribution that has the same min and max as your data. I haven't tried this personally, but I have seen others do it and it makes sense.
2. We applied cutout after normalization. That's mainly because we wanted the mean to already be 0 though. If you are using uniform noise then it probably doesn't matter.
3. I don't know if the order of operations matters, but in all applications of cutout that I have seen people applied it last, after all other augmentations.

Hope that helps!
